### PR TITLE
[BEAM-151] Move maven archetypes build order to be after runners

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
sdks/java/maven-archtypes has several dependencies on the DataflowPipelineRunner. Until these are refactored out or a released artifact exists, we need to modify the build order.

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @davorbonaci

File: sdks/java/pom.xml [hunk 1]
```diff
@@ -36,7 +36,6 @@
 
   <modules>
     <module>core</module>
-    <module>maven-archetypes</module>
```
Comment (davorbonaci) [inline sdks/java/pom.xml]: Don't remove, comment out, along with the same description as above.

**Comment by davorbonaci** [comment]
LGTM for self-merge (see one comment).

**Comment by jbonofre** [comment]
LGTM as well. When the refactoring will be done, we will use the sdk reactor instead of the main one.

File: sdks/java/pom.xml (hunk 1 above)
Comment (lukecwik) [inline sdks/java/pom.xml]: Done

**Pull request closed without merging**